src/grpc/infer_handler.h

```diff
@@ -1168,9 +1246,23 @@ InferHandler<
 
     while (cq_->Next(&tag, &ok)) {
       State* state = static_cast<State*>(tag);
+      if (state->step_ == Steps::WAITING_NOTIFICATION) {
+        State* state_wrapper = state;
+        state = state_wrapper->state_ptr_;
+        state->context_->SetReceivedNotification(true);
```

Comment:
Is there any way for `state_wrapper->state_ptr_` to be prematurely cleaned up / nullptr / invalid / etc.?

Reply:
We are preserving the state object's lifetime till all the events and callbacks have been processed. We are not going to release the state until all these conditions are met.